src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp

```diff
@@ -252,16 +254,95 @@ void YARPRobotStatePublisherModule::onRead(yarp::rosmsg::sensor_msgs::JointState
         sizeOfTFFrames = model.getNrOfFrames();
     }
 
-    for (size_t frameIdx=0; frameIdx < sizeOfTFFrames; frameIdx++)
+    if (m_treeType == "SHALLOW")
     {
-        if(m_baseFrameIndex == frameIdx)    // skip self-tranform
-            continue;
-
-        iDynTree::Transform base_H_frame = m_kinDynComp.getRelativeTransform(m_baseFrameIndex, frameIdx);
-        iDynTree::toYarp(base_H_frame.asHomogeneousTransform(), m_buf4x4);
-        m_iframetrans->setTransform(m_tfPrefix + model.getFrameName(frameIdx),
-                                    m_tfPrefix + model.getFrameName(m_baseFrameIndex),
-                                    m_buf4x4);
+        // In shallow mode, we publish the position of each frame of the robot w.r.t. to the base frame of the robot
+        for (size_t frameIdx=0; frameIdx < sizeOfTFFrames; frameIdx++)
+        {
+            if(m_baseFrameIndex == frameIdx)    // skip self-tranform
+                continue;
+
+            iDynTree::Transform base_H_frame = m_kinDynComp.getRelativeTransform(m_baseFrameIndex, frameIdx);
+            iDynTree::toYarp(base_H_frame.asHomogeneousTransform(), m_buf4x4);
+            m_iframetrans->setTransform(m_tfPrefix + model.getFrameName(frameIdx),
+                                        m_tfPrefix + model.getFrameName(m_baseFrameIndex),
+                                        m_buf4x4);
+        }
+    }
+    else
```

Comment:
It's quite common to make a mistake in the configuration file. As it is, mistakes are treated as `DEEP`, and it could be problematic if configuration files start being committed in downstream repos with errors.

Can you please put this branch in a `if (m_treeType == "DEEP")` and add an `else` that returns false notifying a configuration error?